### PR TITLE
Issue #396 add "executable" to supplementary style guidelines

### PR DESCRIPTION
**Pull request by rolfedh**
References #396

### CODE REVIEW

**Review by mportman12 — APPROVED** [review]
LGTM

File: supplementary_style_guide/glossary_terms_conventions/general_conventions/e.adoc [hunk 1]
```diff
@@ -168,6 +168,26 @@ If referring to the program, use "Emacs", for example, "Source-Navigator support
 
 *See also*:
 
+[[executable]]
+==== image:images/yes.png[yes] Executable (adjective)
+*Description*: _Executable_ describes a file an operating system can run as a program. For directories, having executable permission allows you to navigate into the directory with the `cd` command and list the files it contains.
```
Comment (mmuehlfeldRH) [inline supplementary_style_guide/glossary_terms_conventions/general_conventions/e.adoc:173]: The last part is technically not correct. If a directory contains only the x bit, then you cannot list the files in it.
Try it: 
```
$ mkdir test
$ chmod 0111      # (sets only the x bits for user/group/other, and none of the other bits)
$ ls test/
ls: cannot open directory 'test/': Permission denied
```

You set only the x bit (without r) on a directory if the path to a file in this directory is known but you don't want that a user/service/... can browse the directory (of course, this requires the r bit on the file).

If you drop "and list the files it contains" from the sentence, it's correct.

Comment (rolfedh) [reply supplementary_style_guide/glossary_terms_conventions/general_conventions/e.adoc]: Nice catch, @mmuehlfeldRH! Thanks!

**Review by mportman12 — APPROVED** [review]
LGTM

File: supplementary_style_guide/glossary_terms_conventions/general_conventions/e.adoc [hunk 2]
```diff
@@ -168,6 +168,26 @@ If referring to the program, use "Emacs", for example, "Source-Navigator support
...
+
+*Use it*: yes
+
+*Incorrect forms*:
+
+*See also*:
+
+[[executable-file]]
+==== image:images/yes.png[yes] Executable file (noun)
+*Description*: When it runs, an _executable file_ or _executable program_ performs specific operations based on its coded instructions. Write _executable file_ on the first use and, optionally, _executable_ thereafter.
```
Comment (bergerhoffer) [inline supplementary_style_guide/glossary_terms_conventions/general_conventions/e.adoc:183]: I think when @bredamc proposed this, she proposed putting "executable" in parentheses after the first use of "executable program".

Should we also consider adding a separate entry for "executable" as a noun?

**Review by mportman12 — DISMISSED** [review]
LGTM

**Comment by rolfedh** [comment]
Thanks for your review, @bergerhoffer. I've updated and simplied the logic of these two entries.

File: supplementary_style_guide/glossary_terms_conventions/general_conventions/e.adoc [hunk 3]
```diff
@@ -168,6 +168,26 @@ If referring to the program, use "Emacs", for example, "Source-Navigator support
 
 *See also*:
 
+[[executable]]
```
Comment (bergerhoffer) [inline supplementary_style_guide/glossary_terms_conventions/general_conventions/e.adoc:171]: This produces a warning about duplicate IDs in use. Can you change this one to add -adj?

```suggestion
[[executable-adj]]
```

File: supplementary_style_guide/glossary_terms_conventions/general_conventions/e.adoc [hunk 4]
```diff
@@ -168,6 +168,26 @@ If referring to the program, use "Emacs", for example, "Source-Navigator support
...
+
+*Use it*: yes
+
+*Incorrect forms*:
+
+*See also*:
+
+[[executable]]
+==== image:images/caution.png[with caution] executable (noun)
+*Description*: When it runs, an _executable_ performs specific operations based on its coded instructions. Write _executable file (executable)_ on the first use.
```
Comment (bergerhoffer) [inline supplementary_style_guide/glossary_terms_conventions/general_conventions/e.adoc:183]: We use quotes when we tell people what to write.

```suggestion
*Description*: When it runs, an _executable_ performs specific operations based on its coded instructions. Write "executable file (executable)" on the first use.
```

File: supplementary_style_guide/glossary_terms_conventions/general_conventions/e.adoc [hunk 5]
```diff
@@ -168,6 +168,26 @@ If referring to the program, use "Emacs", for example, "Source-Navigator support
...
+
+*Use it*: with caution
+
+*Incorrect forms*: "executable" alone on first use
```
Comment (bergerhoffer) [inline supplementary_style_guide/glossary_terms_conventions/general_conventions/e.adoc:187]: Our incorrect forms field is usually just a single term with no additional guidance. I'd say to just remove this

**Comment by rolfedh** [comment]
Thanks, @bergerhoffer. Updated.

**Review by bergerhoffer — APPROVED** [review]
LGTM!

**Review by mportman12 — APPROVED** [review]
LGTM

**Pull request merged**